src/main.rs

```diff
@@ -262,6 +282,183 @@ fn _main() -> PaziResult {
+            );
+            PaziResult::Success
+        }
+        Err(e) => {
+            println!("pazi: error adding directory: {}", e);
+            PaziResult::Error
+        }
+    }
+}
+
+fn handle_jump(cmd: &ArgMatches) -> PaziResult {
+    let mut frecency = load_frecency();
+    let res;
+
+    { // once non-lexical-lifetimes hits stable, remove these braces
+        let mut matches = match cmd.value_of("dir_target") {
+            Some(to) => {
+                env::current_dir()
+                    .map(|cwd| {
+                        frecency.maybe_add_relative_to(cwd, to);
+                    })
+                .unwrap_or(()); // truly ignore failure to get cwd
+                frecency.directory_matches(to)
+            }
+            None => frecency.items_with_frecency(),
+        };
+
+        res = if cmd.is_present("interactive") {
+            let stdout = termion::get_tty().unwrap();
+            match interactive::filter(matches, std::io::stdin(), stdout) {
+                Ok(el) => {
+                    print!("{}", el);
+                    PaziResult::SuccessDirectory
+                }
+                Err(interactive::FilterError::NoSelection) => {
+                    // early return since no selection arbitrarily implies not trimming non-existent
+                    // paths. The early return skips the save_to_disk below
+                    return PaziResult::ErrorNoInput;
+                }
+                Err(e) => {
+                    println!("{}", e);
+                    return PaziResult::Error;
+                }
+            }
+        } else if let Some((path, _)) = matches.next() {
+            print!("{}", path);
+            PaziResult::SuccessDirectory
+        } else {
+            PaziResult::Error
+        };
+    };
+
+    if let Err(e) = frecency.save_to_disk() {
+        // leading newline in case it was after a 'print' above
+        println!("\npazi: error saving db changes: {}", e);
+        return PaziResult::Error;
+    }
+    res
+}
+
+fn handle_visit(cmd: &ArgMatches) -> PaziResult {
+    let dir = match cmd.value_of("dir_target") {
+        Some(dir) => dir,
+        None => {
+            println!("visit: visit requires a directory target to visit");
+            return PaziResult::Error;
+        }
+    };
+
+    let mut frecency = load_frecency();
+    frecency.visit(dir.to_string());
+
+    match frecency.save_to_disk() {
+        Ok(_) => {
+            return PaziResult::Success;
+        }
+        Err(e) => {
+            println!("pazi: error adding directory: {}", e);
+            return PaziResult::Error;
+        }
+    }
+}
+
+fn handle_print_frecency(cmd: &ArgMatches) -> PaziResult {
+    let mut frecency = load_frecency();
+
+    let matches = match cmd.value_of("dir_target") {
+        Some(to) => {
+            env::current_dir()
+                .map(|cwd| {
+                    frecency.maybe_add_relative_to(cwd, to);
```

Comment:
Do we want this behavior here? It makes sense for `jump` (to help jumpstart (heh) a small database) but for `view` I'm not so sure.